Drift/Private/DriftMessageQueue.cpp

```diff
@@ -176,16 +176,25 @@ void FDriftMessageQueue::GetMessages()
                 fetchDelay = DEFAULT_FETCH_THROTTLE_DELAY_SECONDS;
                 return;
             }
-            const auto oldLastMessage = lastMessageNumber;
+            TArray<FMessageQueueEntry> Messages;
             for (const auto& queue : messages.queues)
             {
                 for (const auto& message : queue.Value)
                 {
-                    ProcessMessage(queue.Key, message);
-                    
-                    lastMessageNumber = FMath::Max(lastMessageNumber, message.message_number);
+                    Messages.Add(message);
                 }
             }
+            Messages.Sort([](const FMessageQueueEntry& lhs, const FMessageQueueEntry& rhs){return lhs.message_number < rhs.message_number; });
+            const auto oldLastMessage = lastMessageNumber;
+            for (const auto& Message: Messages)
+            {
+                ProcessMessage(Message.queue, Message);
+            }
+            lastMessageNumber = oldLastMessage;
```

Comment:
Why copy the value back here?

Reply:
`ProcessMessage()` doesn't modify `lastMessageNumber`

Reply:
Yeah, that makes no sense whatsoever and I can't even recall what I was thinking there (if anything) :P